src/components/Hero/HeroIncome.jsx

```diff
@@ -40,14 +42,81 @@ const transactions = [
 
 const type = "income";
 
-const handleDelete = (id) => {
-  // Implementacja usuwania transakcji
-};
-
 function HeroIncome() {
+  const [transactions, setTransactions] = useState(initialTransactions);
+
+  const handleAddTransaction = async (newTransaction) => {
+    try {
+      // BACKEND usunąć console.log
+      console.log("Adding transaction:", newTransaction);
+      // BACKEND skontrolować endpoint
+      const response = await fetch(
+        "https://kapusta-server.onrender.com/api/transactions/income",
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(newTransaction),
+        }
+      );
+
+      if (!response.ok) {
+        throw new Error("Failed to add transaction");
+      }
+
+      // Pobieram zaktualizowane dane z serwera
+      const addedTransaction = await response.json();
+
+      // BACKEND usunąć console.log
+      console.log("Transaction added:", addedTransaction);
+      // Aktualizuje stan
+      setTransactions([...transactions, addedTransaction]);
+    } catch (error) {
+      console.error("Error adding transaction:", error);
+    }
+  };
+
+  const handleClearTransactions = () => {
+    // BACKEND usunąć console.log
+    console.log("Clearing transactions");
+    setTransactions([]);
+  };
```

Comment:
tutaj to samo
ten clear nie ma czyścić tabelki transakcji tylko zawartość formularza transakcji
jeśli obie funkcje są identyczne czyli czyszczenie formularza to nie jestem pewien czy nie możesz jej podpiąć bezpośrednio w form transactions do tego buttona bo niezależnie gdzie to robi to samo

Reply:
poprawione